### ISSUE TITLE
Crash in backward propagation

### ISSUE THREAD

**Issue by zhaoBowen612** [pytorch/pytorch#84511]
### 🐛 Describe the bug

import numpy as np
import torch
device = torch.device("mps")
conv = torch.nn.Conv1d(in_channels=1, out_channels=1, kernel_size=3).to(device)

data = torch.tensor(np.random.uniform(size=(1, 176, 1)), dtype=torch.float32).to(device)
optimizer = torch.optim.Adam(conv.parameters(), lr=0.1)
x = data.permute(0, 2, 1).to(memory_format=torch.contiguous_format)
# x = data.permute(0, 2, 1). # this may also cause crash
out = torch.sum(conv(x))
loss = torch.nn.MSELoss()(out, torch.zeros_like(out))
optimizer.zero_grad()
out.backward()
optimizer.step()

### Versions

Collecting environment information...
PyTorch version: 1.13.0.dev20220902
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.5.1 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: version 3.23.2
Libc version: N/A

Python version: 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:25:14)  [Clang 12.0.1 ] (64-bit runtime)
Python platform: macOS-12.5.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.5
[pip3] torch==1.13.0.dev20220902
[pip3] torchaudio==0.13.0.dev20220902
[pip3] torchvision==0.14.0.dev20220902
[conda] torch                     1.13.0.dev20220902          pypi_0    pypi
[conda] torchaudio                0.13.0.dev20220902          pypi_0    pypi
[conda] torchvision               0.14.0.dev20220902          pypi_0    pypi

cc @ezyang @gchanan @zou3519 @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @kulinseth

**Comment by zhaoBowen612** [comment]
@kulinseth @albanD

**Comment by zhaoBowen612** [comment]
Are there other people who can also reproduce this issue?

**Comment by zou3519** [comment]
tentatively marking hi-pri for crash but this needs reproduction

**Comment by soulitzer** [comment]
Slightly smaller repro:
```python
import torch

device = torch.device("mps")
conv = torch.nn.Conv1d(in_channels=1, out_channels=1, kernel_size=3).to(device)

# Errors
data = torch.rand(1, 176, 1, dtype=torch.float32)
x = data.permute(0, 2, 1).contiguous().to(device)

# Does not error
# x = torch.rand(1, 1, 176, dtype=torch.float32, device=device)

conv(x).sum().backward()
```

**Comment by zhaoBowen612** [comment]
> 

thank you very much

**Comment by kulinseth** [comment]
Thanks @soulitzer for the repro, I am able to locally reproduce the crash:
```
failed assertion `Source and weight input channels mismatch'
```
cc. @DenisVieriu97
